src/yunohost/firewall.py

```diff
@@ -203,7 +203,13 @@ def firewall_reload(skip_upnp=False):
     # Check if SSH port is allowed
     ssh_port = _get_ssh_port()
     if ssh_port not in firewall_list()['opened_ports']:
-        firewall_allow('TCP', ssh_port, no_reload=True)
+        _run_service_command('stop', 'ssh')
+        _run_service_command('disable', 'ssh')
+        logger.warning(m18n.n('ssh_port_not_opened'))
+    else:
+        _run_service_command('enable', 'ssh') 
+        _run_service_command('start', 'ssh')       
```

Comment:
Uhhhh, I'm not particularly against allowing the user to close the SSH port, but why stop and disable the SSH service ? :|

Reply:
Yes and no, i dunno, this is a touchy piece of code ... To me asking to close port 22 should result in port 22 being closed, not resulting in the system deciding automatically to stop and disable the SSH service :/ 

Later, a user could be willing to somehow regain SSH access to its server, might end up changing the port in the SSH config, restart the SSH server and then restart the SSH daemon ... until the day its server reboots but the service is "disabled" and the SSH service won't start at boot and the user ends up being locked / requiring physical access to the machine.

That's a specific use case but I'm just not really comfortable with doing this automagically :/ The user can still manually stop and disable the SSH daemon manually (using the yunohost cli or webadmin) if he/she sure about that...

Reply:
yes, but if he wanted to "regain SSH access to its server", he will have to open ssh port, no ? I prefer first : 
- not generating error log for a service which won't be used.
- securing the ssh service

But ok, if you want really me to remove those two lines, it's ok. I just find it unsecure, and really even more touchy than enable it when port is closed. It's my point of view.

Reply:
> because ssh can't work if ssh port is not opened... did I forget something ?

ssh is independent of the firewall. In any case you can access to ssh localy (yes it's it really specific use case but it's possible.

I think too that it's better to keep ssh enabled when the firewall is closed.

Reply:
ok. Let's do a little correction.

Reply:
Well I dunno, thinking a bit more about this, I think what really troubles me is that it seems easy to do the "mistake" of disabling the SSH port for some reason, and then getting locked out of your server ...

Imho the ideal solution would be something like this : 
- allowing the closing of the SSH port only via CLI
- asking for explicit confirmation for this specific port ?

Reply:
mh... yes. Need work. For the first PR, I removed "disable/stop" commands.

Reply:
But I let the warning.